src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -76,7 +78,7 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToAssign = lastShownList.get(index.getZeroBased());
 
-        if (personToAssign.getCcas().isEmpty() && !personToAssign.getRoles().isEmpty()) {
```

Comment:
What does this change do?

Reply:
In the case where there is no CCA and no roles assigned yet this will fail to trigger. 

!personToAssign.getRoles().isEmpty() will be false and no error will be thrown even though there is no CCA and no role should be assigned.

The end result is that a role will be assigned to a person even though they do not have CCA.

Reply:
In either case, there is no need to check for the role portion for the assign command as if there is no CCA, the assign command should already fall through.